admin-console/src/main/webapp/src/pages/ApplicationPage.tsx

```diff
@@ -47,19 +47,19 @@ export const ApplicationPage: React.FunctionComponent = () => {
         setShowModal(false);
         const mutate = newEvent.mutate;
         mutate({
-            id: eventType.id ?? '',
+            id: eventType.id,
             displayName: eventType.displayName ?? '',
             name: eventType.name ?? '',
             description: eventType.description ?? '',
-            applicationId: eventType.applicationId ?? ''
+            applicationId: applicationId
 
         }).then(eventTypesQuery.query);
 
-    }, [ newEvent.mutate, eventType.id, eventType.displayName, eventType.name,
-        eventType.description, eventType.applicationId, eventTypesQuery.query ]);
+    }, [ newEvent.mutate, eventType, applicationId, eventTypesQuery.query ]);
```

Comment:
Using `eventType`.
We are managing the state as immutable. (i.e. we don't modify the variable directly, but with the `setState` function. We create new objects, etc.
So is fine to use this as a whole.